src/lucky/routeable.cr

```diff
@@ -79,12 +82,25 @@ module Lucky::Routeable
       end
       is_root_path = path == ""
       path = "/" if is_root_path
+
+      query_params = {} of String => String
+      {% for opt_param in OPTIONAL_PARAMS %}
+        query_params["{{ opt_param }}"] = {{ opt_param }} unless {{ opt_param }}.nil?
+      {% end %}
+
+      unless query_params.empty?
+        path += "?#{HTTP::Params.encode(query_params)}"
```

Comment:
Should we really escape here?
I am seeing what looks like double escaping when I do
```crystal
      link "Next page", to: Blog::Index.with("2")
```

Reply:
Hmm no that might be weird `link` escaping. This fixes it lol:
```crystal
      raw "<a href=\"#{Blog::Index.path("2")}\">Next page</a>"
```